src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,141 @@
+            this.next = next;
+        }
+
+        public final K getKey() {
+            return key;
+        }
+
+        public final V getValue() {
+            return value;
+        }
+
+        public final String toString() {
+            return key + "=" + value;
+        }
+
+        public final int hashCode() {
+            return Objects.hashCode(key) ^ Objects.hashCode(value);
+        }
+
+        public final V setValue(V newValue) {
+            V oldValue = value;
+            value = newValue;
+            return oldValue;
+        }
+
+        public final boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            }
+            if (o instanceof Map.Entry) {
+                Map.Entry<?,?> e = (Map.Entry<?,?>)o;
+                if (Objects.equals(key, e.getKey())
+                        && Objects.equals(value, e.getValue())) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            growAndTransfer();
+        }
+        int keyHash = (key == null) ? 0 : key.hashCode();
+        int index = Math.abs(keyHash) % table.length;
+        Node<K, V> endNode = table[index];
+        if (endNode == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        while (endNode.next != null) {
+            if (Objects.equals(endNode.getKey(), key)) {
+                endNode.setValue(value);
+                return;
+            }
+            endNode = endNode.next;
+        }
+        if (Objects.equals(endNode.getKey(), key)) {
+            endNode.setValue(value);
+            return;
+        }
+        endNode.next = new Node<>(key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int keyHash = (key == null) ? 0 : key.hashCode();
+        int index = Math.abs(keyHash) % table.length;
+        Node<K, V> helpNode = table[index];
+        while (helpNode != null) {
+            if (Objects.equals(key, helpNode.getKey())) {
+                return helpNode.getValue();
+            }
+            helpNode = helpNode.next;
+        }
         return null;
     }
 
+    @Override
+    public boolean remove(K key) {
+        int keyHash = (key == null) ? 0 : key.hashCode();
+        int index = Math.abs(keyHash) % table.length;
+        Node<K, V> helpNode = table[index];
+        if (Objects.equals(helpNode.getKey(), key)) {
+            table[index] = helpNode.next;
+            return true;
+        }
+        while (!Objects.equals(helpNode.next.getKey(), key)) {
+            helpNode = helpNode.next;
+            if (Objects.equals(helpNode.next.getKey(), key)) {
+                helpNode.next = helpNode.next.next;
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
I dont think it was a part of the task to implement this method, but good job!
And again, you can create your helpNode in one line as I've explained before